src/core/helpers/dates/companiDates.ts

```diff
@@ -0,0 +1,69 @@
+import { DateTimeUnit } from 'luxon';
+import { DateTime } from './luxon';
+
+type CompaniDateType = {
+  _date: DateTime;
+  format: (str: string) => string,
+  toDate: () => Date,
+  isBefore: (date: Date | CompaniDateType | string) => Boolean,
+  startOf: (unit: DateTimeUnit) => CompaniDateType,
+  endOf: (unit: DateTimeUnit) => CompaniDateType,
+}
+
+const CompaniDate = (...args: any[]) => CompaniDateFactory(_formatMiscToCompaniDate(...args));
+
+const CompaniDateFactory = (inputDate: DateTime): CompaniDateType => {
+  const _date = inputDate;
+
+  return ({
+    // GETTER
+    get _date() {
+      return _date;
+    },
+
+    // DISPLAY
+    format(fmt: string) {
+      return _date.toFormat(fmt);
+    },
+
+    toDate() {
+      return this._date.toUTC().toJSDate();
+    },
+
+    // QUERY
+    isBefore(miscTypeOtherDate : Date | CompaniDateType | string) {
+      const otherDate = _formatMiscToCompaniDate(miscTypeOtherDate);
+
+      return _date < otherDate;
+    },
+
+    // MANIPULATE
+    startOf(unit: DateTimeUnit) {
+      return CompaniDateFactory(_date.startOf(unit));
+    },
+
+    endOf(unit: DateTimeUnit) {
+      return CompaniDateFactory(_date.endOf(unit));
+    },
+  });
+};
+
+const _formatMiscToCompaniDate = (...args: any[]) => {
```

Comment:
Que penses-tu de preciser les arguments acceptés, quelque chose comme
```
type authorizedDateType = Date | String | DateTime | CompaniDateType;
const _formatMiscToCompaniDate = (...args: [authorizedDateType, String] | [authorizedDateType]) => {
```

Reply:
(J'aime pas trop mon 'authorizedDateType' mais je n'arrive pas a trouve un meilleur nom)

Reply:
Que penses-tu de `DateTypes`?

Reply:
Ici quand je type args avec : `[] | [DateTypes] | [DateTypes, string]` j'ai l'erreur sur `args[0]?._date instanceof DateTime` : `Property '_date' does not exist on type 'Date'` j'ai l'impression que ca vient de l'union, typescript voudrait que chaque type de l'union ait le champ _date: DateTime

Reply:
Done